airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/DestinationConsumerStrategy.java

```diff
@@ -1,39 +0,0 @@
-/*
- * MIT License
- *
- * Copyright (c) 2020 Airbyte
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in all
- * copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- * SOFTWARE.
- */
-
-package io.airbyte.integrations.destination;
-
-import io.airbyte.integrations.base.DestinationConsumer;
-import io.airbyte.protocol.models.AirbyteMessage;
-import java.util.Map;
-
-/**
- * DestinationConsumer configured with a DestinationWriteContext configuration object to slightly
- * change their behavior at runtime.
- */
-public interface DestinationConsumerStrategy extends DestinationConsumer<AirbyteMessage> {
-
-  void setContext(Map<String, DestinationWriteContext> configs) throws Exception;
```

Comment:
I was having a lot of trouble following the pattern where we were using `setContext`. I think chain of responsibility was overkill for this and ended up separating logic in a way that made it hard for me to follow.

Reply:
I added all that because some comments were made about destinations that may or may not need to write to a temporary table.

Does it seem like with this version, we always consider passing by a tmpTable?

Reply:
yeah. in this version we are just always writing to temp tables. for all of the jdbc destinations that seems to be the pattern we were following anyway.